lib/core-integration/src/Test/Integration/Framework/DSL.hs

```diff
@@ -834,6 +849,56 @@ delegationFee
 delegationFee ctx w = do
     request @ApiFee ctx (Link.getDelegationFee w) Default Empty
 
+-- | Generate an infinite list of addresses for random wallets.
+--
+-- To be typically used as:
+--
+-- >>> take 1 (randomAddresses @n)
+-- [addr]
+randomAddresses
+    :: forall (n :: NetworkDiscriminant). (PaymentAddress n ByronKey)
+    => Mnemonic 12
+    -> [Address]
+randomAddresses mw =
```

Comment:
I am wondering if we should not try to unify *random* and *icarus* functions under one hood. But maybe not, having them standalone maybe incur additional LOC, but it is more clear how they are constructed and hence they are much clearer to understand (for me at least)

Reply:
> I am wondering if we should not try to unify random and icarus functions under one hood. 

I am wondering the same to be honest. So far, I haven't really found a good justification for doing so however, so I am refraining myself ^.^